unity/PingPongGameExample/PingPongGame/Assets/Scripts/NetTest.cs

```diff
@@ -15,42 +15,100 @@
  * limitations under the License.
  */
 
-using System.Collections;
 using System.Collections.Concurrent;
 
 using System.Net.Sockets;
+using System.Net.Http;
+using System.Net.NetworkInformation;
 using System.Diagnostics;
 using System.Threading;
-using System.Threading.Tasks;
 
 using System;
 
 public class NetTest
 {
-  public class HostAndPort
+  public enum TestType
+  {
+    PING = 0,
+    CONNECT = 1,
+  };
+
+  public class Site
   {
     public string host;
     public int port;
-    public double lastPingMs = 0f;
+    public string L7Path; // This may be load balanced.
+    public double lastPingMs;
+
+    public TestType testType;
+
+    int idx;
+    int size;
+    public double[] samples;
+
+    public double average;
+    public double stddev;
+
+    public Site(TestType testType = TestType.CONNECT, int numSamples = 5)
+    {
+      this.testType = testType;
+      samples = new double[numSamples];
+    }
+
+    public void addSample(double time)
+    {
+      samples[idx] = time;
+      idx++;
+      if (size < samples.Length) size++;
+      idx = idx % samples.Length;
+    }
+
+    public void recalculateStats()
```

Comment:
Is there any appeal in keeping a running average and std dev for all samples as well? Developer can have both a moving window view and a view of stats for the entire usage of the app?

Reply:
Here is a RollingAverage class used by the computervision library for displaying Cloud/Edge and Full-Process/Network-only stats: 
https://github.com/mobiledgex/edge-cloud-sampleapps/blob/master/android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/RollingAverage.java

Reply:
Client side only. Future. A backend endpoint would love histograms though.